client/landing/stepper/hooks/use-site-migration-key.ts

```diff
@@ -28,19 +28,17 @@ const getMigrationKey = async ( siteId: number ): Promise< ApiResponse > => {
 	);
 };
 
-type Options = Pick< UseQueryOptions, 'enabled' >;
+type Options = {
+	enabled?: UseQueryOptions[ 'enabled' ];
+	retry?: UseQueryOptions[ 'retry' ];
+};
 
 export const useSiteMigrationKey = ( siteId?: number, options?: Options ) => {
 	return useQuery( {
 		queryKey: [ 'site-migration-key', siteId ],
 		queryFn: () => getMigrationKey( siteId! ),
-		retry: ( failureCount: number, error: Error ) => {
-			if ( isWhiteLabeledPluginEnabled() && failureCount >= 20 ) {
-				throw error;
```

Comment:
Maybe we should keep the `failureCount` check to avoid retrying indefinitely if something fails?

Reply:
The total attempts is configured here https://github.com/Automattic/wp-calypso/pull/94854/files#diff-087f2598d234871cab198d6f6ad9a6008cb7b9d7673d0a1a90336bcfb880fa73R215

Reply:
Bit more context.
retry is a mult typed field (?!)  that can receive false | number |  or a custom function. 
If we pass a number, the query will use it as total of attempts.

Reply:
Oh, missed that. Thanks for explaining.